ui/src/app/views/registration-dashboard.tsx

```diff
@@ -169,25 +178,34 @@ export class RegistrationDashboard extends React.Component<RegistrationDashboard
     super(props);
     this.state = {
       trainingWarningOpen: !props.firstVisitTraining,
-      taskCompletionMap: new Map<number, boolean>(),
-      showRefreshButton: false
+      showRefreshButton: false,
+      bypassActionComplete: false,
+      bypassInProgress: false,
     };
-    this.state.taskCompletionMap.set(0, props.twoFactorAuthCompleted);
-    this.state.taskCompletionMap.set(1, props.trainingCompleted);
-    this.state.taskCompletionMap.set(2, props.eraCommonsLinked);
-    this.state.taskCompletionMap.set(3, props.dataUseAgreementCompleted);
   }
 
   componentDidMount() {
     this.setState({showRefreshButton: false});
   }
 
+  get taskCompletionList(): Array<boolean> {
+    const list = getRegistrationTasks().map((config) => {
+      return this.props[config.completionPropsKey] as boolean;
+    });
+    return list;
+  }
+
+  allTasksCompleted(): boolean {
+    return this.taskCompletionList.reduce((acc, val) => acc && val);
```

Comment:
opt_nit: I'm not sure I like it any better than what you have, but one alternative I came up with:

`this.taskCompletionList.every(v => v);` or `this.taskCompletionList.every(_.identity)`